cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -297,8 +331,30 @@ public void performAction(HttpRequest request, HttpResponder responder,
                             @PathParam("type") String type,
                             @PathParam("id") String id,
                             @PathParam("action") String action) throws Exception {
+    performActionHelper(request, responder, namespaceId, appId, ApplicationId.DEFAULT_VERSION, type, id, action);
+  }
+
+  @POST
+  @Path("/apps/{app-id}/versions/{version-id}/{type}/{id}/{action}")
+  public void performVersionAction(HttpRequest request, HttpResponder responder,
```

Comment:
Thought it would make it clearer that these methods are for versioned API when they get called.